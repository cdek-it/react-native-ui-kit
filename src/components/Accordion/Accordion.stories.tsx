import type { Meta, StoryObj } from '@storybook/react'
import {
  IconDiamond,
  IconList,
  IconSquare,
  IconUser,
} from '@tabler/icons-react-native'

import { memo } from 'react'
import { View } from 'react-native'

import { Body } from '../Typography'

import { Accordion, type AccordionProps } from './Accordion'

const Icons = { undefined, IconUser, IconList, IconSquare }
const Extras = {
  undefined,
  IconDiamond: <IconDiamond height={21} width={21} />,
}

const TestAccordion = memo<Omit<AccordionProps, 'childern'>>((props) => {
  return (
    <Accordion {...props}>
      <View style={{ padding: 24, backgroundColor: 'blanchedalmond' }}>
        <Body>Это контент аккордеона</Body>
      </View>
    </Accordion>
  )
})

const meta: Meta<typeof Accordion> = {
  title: 'Accordion',
  component: TestAccordion,
  args: {
    title: 'Accordion',
    Icon: undefined,
    titleExtra: undefined,
    withSeparator: false,
  },
  argTypes: {
    Icon: { control: 'select', options: Object.keys(Icons), mapping: Icons },
    title: { control: 'text' },
    titleExtra: {
      control: 'select',
      options: Object.keys(Extras),
      mapping: Extras,
    },
    withSeparator: { control: 'boolean' },
    disabled: { control: 'boolean' },
  },
  decorators: [
    (Story) => (
      <View
        style={{ padding: 16, alignContent: 'center', alignItems: 'center' }}
      >
        <Story />
      </View>
    ),
  ],
}

export default meta

type Story = StoryObj<typeof Accordion>

export const AccordionStory: Story = {
  args: { title: 'Accordion' },
  argTypes: {},
  parameters: {
    notes: `
      #### Акордеон - полный набор свойств

      <Accordion
        title="Menu item title"
        Icon=IconUser,
        extra=IconDiamond,
        isExpanded
      >
        // Content
      </Accordion>
  `,
  },
}
