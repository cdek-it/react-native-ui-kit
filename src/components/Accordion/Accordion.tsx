import { type Icon, IconChevronRight } from '@tabler/icons-react-native'
import React, { useCallback, useState } from 'react'
import { View, Text, Pressable, type LayoutChangeEvent } from 'react-native'
import Animated, {
  interpolate,
  useAnimatedStyle,
  useSharedValue,
  withTiming,
} from 'react-native-reanimated'

import type { ViewProps } from 'react-native-svg/lib/typescript/fabric/utils'

import { makeStyles } from '../../utils/makeStyles'

export interface AccordionProps extends ViewProps {
  // Иконка слева от заголовка
  readonly Icon?: Icon
  // Текст заголовка
  readonly title: string
  // Состояние при первом рендере. true - контент раскрыт, false - контент свернут
  readonly isInitiallyExpanded?: boolean
  // Наличие разделитель. Когда true, верхняя граница заголовка становится разделителем
  readonly withSeparator?: boolean
  // Отключенное состояние. Когда true компонент рендерится в соответствии со значением isInitiallyExpanded и не реагирует на нажатия
  readonly disabled?: boolean
  // Дополнительный элемент справа от заголовка
  readonly titleExtra?: React.ReactNode
  readonly children: React.ReactNode
}

/**
 * Компонент Гармошка - умеет скрывать и раскрывать контент по нажатию на заголовок
 * @param title - Текст заголовка
 * @param Icon - Иконка слева от заголовка
 * @param isInitiallyExpanded - Состояние при первом рендере. true - контент раскрыт, false - контент свернут
 * @param extra - Дополнительный элемент справа от заголовка
 * @param withSeparator - Наличие разделитель. Когда true, верхняя граница заголовка становится разделителем
 * @param disabled - Отключенное состояние. Когда true компонент рендерится в соответствии со значением isInitiallyExpanded и не реагирует на нажатия
 * @link https://www.figma.com/design/4TYeki0MDLhfPGJstbIicf/UI-kit-PrimeFace--DS-?node-id=1207-1852&m=dev
 */
export const Accordion: React.FC<AccordionProps> = ({
  Icon,
  title,
  isInitiallyExpanded: initiallyExpanded = false,
  withSeparator = false,
  disabled = false,
  titleExtra,
  testID,
  children,
  ...rest
}) => {
  const styles = useStyles()

  const contentHeight = useSharedValue(0)
  const contentOpenFraction = useSharedValue(initiallyExpanded ? 1 : 0)
  const [isExpanded, setIsExpanded] = useState(initiallyExpanded)

  const onLayout = useCallback(
    (event: LayoutChangeEvent) => {
      contentHeight.value = event.nativeEvent.layout.height
    },
    [contentHeight]
  )

  const toggle = useCallback(() => {
    contentOpenFraction.value = withTiming(
      contentOpenFraction.value > 0 ? 0 : 1
    )
    setIsExpanded((value) => !value)
  }, [contentOpenFraction])

  const arrowAnimatedStyle = useAnimatedStyle(() => ({
    transform: [
      {
        rotate: `${interpolate(contentOpenFraction.value, [0, 1], [0, 90])}deg`,
      },
    ],
  }))

  const contentAnimatedStyle = useAnimatedStyle(() => ({
    height: interpolate(
      contentOpenFraction.value,
      [0, 1],
      [0, contentHeight.value]
    ),
    opacity: contentOpenFraction.value,
  }))

  return (
    <View
      style={[styles.component, withSeparator ? styles.separator : {}]}
      testID={testID || AccordionTestIds.component}
      {...rest}
    >
      <Pressable
        accessible
        accessibilityLabel={title}
        accessibilityRole='button'
        accessibilityState={isExpanded ? { expanded: true } : {}}
        disabled={disabled}
        style={[styles.header, disabled ? styles.disabled : {}]}
        testID={AccordionTestIds.header}
        onPress={toggle}
      >
        <Animated.View
          style={arrowAnimatedStyle}
          testID={AccordionTestIds.arrow}
        >
          <IconChevronRight {...styles.icon} />
        </Animated.View>
        {Icon ? <Icon {...styles.icon} testID={AccordionTestIds.icon} /> : null}
        <Text style={styles.title}>{title}</Text>
        {titleExtra ? (
          <View testID={AccordionTestIds.titleExtra}>{titleExtra}</View>
        ) : null}
      </Pressable>

      <Animated.View
        accessibilityElementsHidden={!isExpanded}
        importantForAccessibility={isExpanded ? 'yes' : 'no-hide-descendants'}
        style={[styles.contentAnimated, contentAnimatedStyle]}
        testID={AccordionTestIds.content}
      >
        <View
          style={styles.contentWrapper}
          testID={AccordionTestIds.contentWrapper}
          onLayout={onLayout}
        >
          {children}
        </View>
      </Animated.View>
    </View>
  )
}

export const AccordionTestIds = {
  component: 'Accordion',
  header: 'Header',
  arrow: 'Arrow',
  icon: 'TitleIcon',
  titleExtra: 'Extra',
  content: 'Content',
  contentWrapper: 'ContentWrapper',
  separator: 'Separator',
}

const useStyles = makeStyles(({ theme }) => ({
  component: { width: '100%' },
  header: {
    paddingVertical: theme.Panel.Accordion.accordionHeaderPaddingTopBottom,
    gap: 7,
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: theme.Panel.Accordion.accordionHeaderBg,
  },
  icon: {
    width: 17.5,
    height: 17.5,
    color: theme.Panel.Accordion.accordionHeaderTextColor,
  },
  title: {
    fontSize: 15.75,
    includeFontPadding: false,
    verticalAlign: 'middle',
    fontWeight: 700,
    color: theme.Panel.Accordion.accordionHeaderTextColor,
  },
  contentAnimated: { overflow: 'hidden' },
  contentWrapper: {
    position: 'absolute',
    width: '100%',
    paddingLeft: theme.Panel.Accordion.accordionContentPaddingLeft,
    paddingTop: theme.Panel.Accordion.accordionContentPaddingTop,
    paddingRight: theme.Panel.Accordion.accordionContentPaddingRight,
    paddingBottom: theme.Panel.Accordion.accordionContentPaddingBottom,
  },
  separator: {
    borderTopColor: theme.Panel.Accordion.accordionHeaderBorderColor,
    borderTopWidth: 1,
  },
  disabled: { mixBlendMode: 'luminosity', opacity: 0.6 },
}))
