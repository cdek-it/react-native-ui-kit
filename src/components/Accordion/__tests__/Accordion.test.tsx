import { IconAddressBook, IconUser } from '@tabler/icons-react-native'
import {
  fireEvent,
  render,
  screen,
  waitFor,
  type within,
} from '@testing-library/react-native'
import { Text, View } from 'react-native'

import { Accordion, AccordionTestIds, type AccordionProps } from '../Accordion'

describe('Accordion', () => {
  test('Header elements minimal', () => {
    const props: Omit<AccordionProps, 'children'> = { title: 'Accordion' }
    const contentText = 'This is Accordion content'
    const { toJSON } = render(
      <Accordion {...props}>
        <Text>{contentText}</Text>
      </Accordion>
    )

    expect(toJSON()).toMatchSnapshot()

    assertAccordionIsDisplaying(props)
  })

  test('Header elements maximal', () => {
    const props: Omit<AccordionProps, 'children'> = {
      Icon: IconUser,
      title: 'Accordion',
      titleExtra: <IconAddressBook />,
      withSeparator: true,
    }
    const contentText = 'This is Accordion content'
    const { toJSON } = render(
      <Accordion {...props}>
        <Text>{contentText}</Text>
      </Accordion>
    )

    expect(toJSON()).toMatchSnapshot()

    assertAccordionIsDisplaying(props)
  })

  test('Content toggle', async () => {
    const title = 'Title'
    render(
      <Accordion isInitiallyExpanded title={title}>
        <View style={{ height: 100 }} />
      </Accordion>
    )

    const includeHidden = { includeHiddenElements: true }

    fireEvent(
      screen.getByTestId(AccordionTestIds.contentWrapper, includeHidden),
      'layout',
      { nativeEvent: { layout: { height: 100, width: 200, x: 0, y: 0 } } }
    )

    await waitFor(() =>
      expect(
        screen.getByTestId(AccordionTestIds.content, includeHidden)
      ).toHaveAnimatedStyle({ height: 100, opacity: 1 })
    )

    expect(screen.getByTestId(AccordionTestIds.arrow)).toHaveAnimatedStyle({
      transform: [{ rotate: `90deg` }],
    })

    fireEvent.press(screen.getByText(title))

    await waitFor(() =>
      expect(
        screen.getByTestId(AccordionTestIds.content, includeHidden)
      ).toHaveAnimatedStyle({ height: 0, opacity: 0 })
    )

    expect(screen.getByTestId(AccordionTestIds.arrow)).toHaveAnimatedStyle({
      transform: [{ rotate: '0deg' }],
    })
  })
})

type TestInstance = ReturnType<typeof within>

const assertAccordionIsDisplaying = (
  props: Omit<AccordionProps, 'children'>,
  testInstance?: TestInstance
) => {
  const instance = testInstance ?? screen

  expect(instance.getByText(props.title)).toBeVisible()

  if (props.Icon) {
    expect(instance.getAllByTestId(AccordionTestIds.icon).length).toBeTruthy()
  } else {
    expect(instance.queryByTestId(AccordionTestIds.icon)).toBeFalsy()
  }

  if (props.titleExtra) {
    expect(instance.getByTestId(AccordionTestIds.titleExtra)).toBeVisible()
  } else {
    expect(instance.queryByTestId(AccordionTestIds.titleExtra)).toBeFalsy()
  }

  if (props.withSeparator) {
    expect(instance.getByTestId(AccordionTestIds.component)).toHaveStyle({
      borderTopWidth: 1,
    })
  } else {
    expect(instance.getByTestId(AccordionTestIds.component)).not.toHaveStyle({
      borderTopWidth: 1,
    })
  }
}
