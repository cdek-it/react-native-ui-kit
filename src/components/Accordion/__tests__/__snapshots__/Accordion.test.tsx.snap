// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Header elements maximal 1`] = `
<View
  style={
    [
      {
        "width": "100%",
      },
      {
        "borderTopColor": "#dbdada",
        "borderTopWidth": 1,
      },
    ]
  }
  testID="Accordion"
>
  <View
    accessibilityLabel="Accordion"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0.0001)",
          "flexDirection": "row",
          "gap": 7,
          "paddingVertical": 14,
        },
        {},
      ]
    }
    testID="Header"
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {
            "transform": [
              {
                "rotate": "0deg",
              },
            ],
          },
        }
      }
      style={
        {
          "transform": [
            {
              "rotate": "0deg",
            },
          ],
        }
      }
      testID="Arrow"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={17.5}
        bbWidth={17.5}
        fill="none"
        focusable={false}
        height={17.5}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="rgba(0, 0, 0, 0.6000)"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 17,
              "width": 17,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={17.5}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 2566914048,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
        >
          <RNSVGPath
            d="M9 6l6 6l-6 6"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 2566914048,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={17.5}
      bbWidth={17.5}
      fill="none"
      focusable={false}
      height={17.5}
      meetOrSlice={0}
      minX={0}
      minY={0}
      stroke="rgba(0, 0, 0, 0.6000)"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 17,
            "width": 17,
          },
        ]
      }
      testID="TitleIcon"
      vbHeight={24}
      vbWidth={24}
      width={17.5}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 2566914048,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      >
        <RNSVGPath
          d="M8 7a4 4 0 1 0 8 0a4 4 0 0 0 -8 0"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 2566914048,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="TitleIcon"
        />
        <RNSVGPath
          d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 2566914048,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="TitleIcon"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        {
          "color": "rgba(0, 0, 0, 0.6000)",
          "fontSize": 15.75,
          "fontWeight": 700,
          "includeFontPadding": false,
          "verticalAlign": "middle",
        }
      }
    >
      Accordion
    </Text>
    <View
      testID="Extra"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={24}
        bbWidth={24}
        fill="none"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "type": 2,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
        >
          <RNSVGPath
            d="M20 6v12a2 2 0 0 1 -2 2h-10a2 2 0 0 1 -2 -2v-12a2 2 0 0 1 2 -2h10a2 2 0 0 1 2 2z"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M10 16h6"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M13 11m-2 0a2 2 0 1 0 4 0a2 2 0 1 0 -4 0"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M4 8h3"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M4 12h3"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M4 16h3"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
  <View
    accessibilityElementsHidden={true}
    collapsable={false}
    importantForAccessibility="no-hide-descendants"
    jestAnimatedStyle={
      {
        "value": {
          "height": 0,
          "opacity": 0,
        },
      }
    }
    style={
      {
        "height": 0,
        "opacity": 0,
        "overflow": "hidden",
      }
    }
    testID="Content"
  >
    <View
      onLayout={[Function]}
      style={
        {
          "paddingBottom": 14,
          "paddingLeft": 24.5,
          "paddingRight": 0,
          "paddingTop": 0,
          "position": "absolute",
          "width": "100%",
        }
      }
      testID="ContentWrapper"
    >
      <Text>
        This is Accordion content
      </Text>
    </View>
  </View>
</View>
`;

exports[`Accordion Header elements minimal 1`] = `
<View
  style={
    [
      {
        "width": "100%",
      },
      {},
    ]
  }
  testID="Accordion"
>
  <View
    accessibilityLabel="Accordion"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0.0001)",
          "flexDirection": "row",
          "gap": 7,
          "paddingVertical": 14,
        },
        {},
      ]
    }
    testID="Header"
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {
            "transform": [
              {
                "rotate": "0deg",
              },
            ],
          },
        }
      }
      style={
        {
          "transform": [
            {
              "rotate": "0deg",
            },
          ],
        }
      }
      testID="Arrow"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={17.5}
        bbWidth={17.5}
        fill="none"
        focusable={false}
        height={17.5}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="rgba(0, 0, 0, 0.6000)"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 17,
              "width": 17,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={17.5}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 2566914048,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
        >
          <RNSVGPath
            d="M9 6l6 6l-6 6"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 2566914048,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        {
          "color": "rgba(0, 0, 0, 0.6000)",
          "fontSize": 15.75,
          "fontWeight": 700,
          "includeFontPadding": false,
          "verticalAlign": "middle",
        }
      }
    >
      Accordion
    </Text>
  </View>
  <View
    accessibilityElementsHidden={true}
    collapsable={false}
    importantForAccessibility="no-hide-descendants"
    jestAnimatedStyle={
      {
        "value": {
          "height": 0,
          "opacity": 0,
        },
      }
    }
    style={
      {
        "height": 0,
        "opacity": 0,
        "overflow": "hidden",
      }
    }
    testID="Content"
  >
    <View
      onLayout={[Function]}
      style={
        {
          "paddingBottom": 14,
          "paddingLeft": 24.5,
          "paddingRight": 0,
          "paddingTop": 0,
          "position": "absolute",
          "width": "100%",
        }
      }
      testID="ContentWrapper"
    >
      <Text>
        This is Accordion content
      </Text>
    </View>
  </View>
</View>
`;
